parsing/parser.mly

```diff
@@ -2388,8 +2381,10 @@ val_longident:
 ;
 constr_longident:
     mod_longident       %prec below_DOT         { $1 }
+  | mod_longident DOT LPAREN COLONCOLON RPAREN  { Ldot($1,"::") }
```

Comment:
Well, it's a bit weird that the qualified syntax is not available for these constructors.

But interpreting `M.[]` as a qualified reference to constructor `[]` would be incompatible with the current interpretation of `M.[e1; ...; en]` (which is a local open).  I now think it was a mistake to have this local open interpretation.  It would have been better to map the expression above to:

````ocaml
  M.(::) e1(M.(::) e2 ..(M.(::) en M.[])
````
(interpreting M.(::) and M.[] as normal qualified references)